### PR TITLE
Implemented Load balancing based on client’s IP

### PR DESCRIPTION
**Pull request by lorenzoc25**

### CODE REVIEW

File: src/load-balancing.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
 import { Middleware } from '../types/middleware';
 
+const ipToNum = (
+  ip: string | null,
+) :number => {
+  if (ip === null) {
+    throw new Error('Error: Unable to get IP.');
+  }
+  return ip.split('0').map((octect, index, array) => parseInt(octect, 10) * (256 ** (array.length - index - 1))).reduce((accumulator, current) => accumulator + current);
```
Comment (xiaoyang-sde) [inline src/load-balancing.ts:9]: I don't think this function is correct...

**Review by xiaoyang-sde — CHANGES_REQUESTED** [review]
Thanks for your contribution! I think we could add a fallback option to select a random upstream if the IP address is missing.

File: src/load-balancing.ts (hunk 1 above)
Comment (lorenzoc25) [reply src/load-balancing.ts]: that was a careless mistake... sorry for that. I've fixed that in the newer version

File: src/load-balancing.ts [hunk 2]
```diff
@@ -1,14 +1,27 @@
 import { Middleware } from '../types/middleware';
 
+const ipToNum = (
+  ip: string | null,
+) :number => {
+  if (ip === null) {
+    return -1;
+  }
```
Comment (xiaoyang-sde) [inline src/load-balancing.ts:8]: I think you've already checked if `ip` is null in line 20.

**Review by xiaoyang-sde — APPROVED** [review]

**Pull request merged**